mantle/cmd/ore/gcloud/update-image.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/coreos/mantle/platform/api/gcloud"
+)
+
+var (
+	cmdUpdate = &cobra.Command{
+		Use:   "update-image",
+		Short: "Update os image",
+		Long:  "Update os image attributes in GCP.",
+		Run:   runUpdateImage,
+	}
+
+	updateImageName        string
+	updateImageFamily      string
+	updateImageDescription string
+)
+
+func init() {
+	cmdUpdate.Flags().StringVar(
+		&updateImageName, "image", "", "GCP image name")
+	cmdUpdate.Flags().StringVar(
+		&updateImageFamily, "family", "",
+		"Updated GCP image family to attach image to")
+	cmdUpdate.Flags().StringVar(
+		&updateImageDescription, "description", "",
+		"The updated description for the image")
+	GCloud.AddCommand(cmdUpdate)
+}
+
+func runUpdateImage(cmd *cobra.Command, args []string) {
+	// Check that the user provided an image
+	if updateImageName == "" {
+		plog.Fatal("Must provide an image name via --image")
+	}
+	// Check that the user provided at least one thing to change
+	if updateImageFamily == "" && updateImageDescription == "" {
+		plog.Fatal("Must provide one of --family or --description")
+	}
+
+	// Create the ImageSpec. Don't worry about passing "" for
+	// family or description. If "" is passed no update will happen.
```

Comment:
Does this mean it's impossible to remove an image from a family?

Reply:
actually I think I stumbled upon a possible answer for this today when I was playing around with trying to get it to deprecate an image. I think there is a way to force sending fields even if they're empty and we need to use that here. I'll experiment some more.

Reply:
to be clear, previously I thought the answer to your question was "yes", but now I'm thinking there is a way.

Reply:
ok I did some testing today with `image.ForceSendFields` and `image.NullFields`. Using either one of those with image family doesn't work. `image.ForceSendFields` sends `""` across the API and the remove validation gives back: 

```
googleapi: Error 400: Invalid value for field 'resource.family': ''. Must be a match 
of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)',
```

So it won't let me set the image family to nothing. Sending null via `image.NullFields` succeeds but doesn't actually remove the family from the image. I'm just going to say it's possible to change an images image family, but it's not possible to disassociate an image from any image family.